Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp

```diff
@@ -647,8 +647,14 @@ void GPUConnectionToWebProcess::createRemoteGPU(WebGPUIdentifier identifier, Ren
     auto* renderingBackend = it->value.get();
 
     auto addResult = m_remoteGPUMap.ensure(identifier, [&] {
-        return IPC::ScopedActiveMessageReceiveQueue { RemoteGPU::create([this] (MediaPlayerIdentifier identifier, Function<void(MediaPlayer&)>&& callback) mutable {
-            this->performWithMediaPlayerOnMainThread(identifier, WTFMove(callback));
+        return IPC::ScopedActiveMessageReceiveQueue { RemoteGPU::create([&] (std::variant<WebCore::MediaPlayerIdentifier, WebKit::RemoteVideoFrameReference> identifier, Function<void(RefPtr<WebCore::VideoFrame>)>&& callback) mutable {
+            return WTF::switchOn(identifier, [&] (WebCore::MediaPlayerIdentifier i) {
+                performWithMediaPlayerOnMainThread(i, [&callback](auto& player) {
```

Comment:
you can't capture by reference here, callback would have gone out of scope prior. Capture with [callback = WTFMove(callback)]

Reply:
And is the call back to `capture()` thread safe? What is GPUConnectToWebProcess thread ?

Reply:
@jyavenard is it really an issue to capture a reference here? `performWithMediaPlayerOnMainThread` is synchronous and the callback runs prior to `performWithMediaPlayerOnMainThread` returning. It uses a semaphore and blocks the calling thread until the callback runs. The semaphore is signaled after the callback completes.

But maybe I should not be relying on the implementation details of `performWithMediaPlayerOnMainThread` and perform the copy as you suggest instead? What do you think?

Reply:
Regarding the thread safety, since `performWithMediaPlayerOnMainThread` uses a BinarySemaphore, I think there are no thread safety considerations. 

However again this is relying on the implementation of `performWithMediaPlayerOnMainThread`. But I don't think `performWithMediaPlayerOnMainThread` thread safety guarantees could be changed at this point without breaking all callers who may rely on the current implementation.